### PR TITLE
MDEXP-641 - Quick-Export: Rewrite/clean up existing and non-changed logic

### PR DESCRIPTION
**Pull request by obozhko-folio**
[MDEXP-641](https://folio-org.atlassian.net/browse/MDEXP-641) - Quick-Export: Rewrite/clean up existing and non-changed logic

## Purpose
Migrate non-affected services according to new approach.

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] Check logging
  - [x] There are no major code smells or security issues

- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [x] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/main/java/org/folio/dataexp/service/QuickExportService.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+public class QuickExportService {
+
+  private final FileDefinitionsService fileDefinitionsService;
+  private final DataExportService dataExportService;
+  private final JobProfileEntityRepository jobProfileEntityRepository;
+  private final ExportIdEntityRepository exportIdEntityRepository;
+  private final JobExecutionEntityRepository jobExecutionEntityRepository;
+
+  public QuickExportResponse postQuickExport(QuickExportRequest quickExportRequest) {
+    var fileDefinition = new FileDefinition().id(UUID.randomUUID()).size(0).fileName(quickExportRequest.getType() + "-quick.csv");
```
Comment (alekGbuz) [inline src/main/java/org/folio/dataexp/service/QuickExportService.java:33]: Hi @obozhko-folio , according MDEXP-677 quick export pattern is quick-export-<export job id>.mrc, please rename fileName to quick-export.csv. <export job id> will be added at file uploading.

File: src/main/java/org/folio/dataexp/service/QuickExportService.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    if (nonNull(uuids)) {
+      var batch = new ArrayList<ExportIdEntity>();
+      uuids.forEach(instanceId -> {
+        var entity = ExportIdEntity.builder().jobExecutionId(fileDefinition
+          .getJobExecutionId()).instanceId(instanceId).build();
+        batch.add(entity);
+      });
+      exportIdEntityRepository.saveAll(batch);
+    } else {
+      log.error("Nothing to export: no uuids provided.");
```
Comment (alekGbuz) [inline src/main/java/org/folio/dataexp/service/QuickExportService.java:64]: Please, add file definition id to the error message.

**Review by siarhei-charniak — APPROVED** [review]

File: src/main/java/org/folio/dataexp/service/QuickExportService.java (hunk 1 above)
Comment (obozhko-folio) [reply src/main/java/org/folio/dataexp/service/QuickExportService.java]: Done.

File: src/main/java/org/folio/dataexp/service/QuickExportService.java (hunk 2 above)
Comment (obozhko-folio) [reply src/main/java/org/folio/dataexp/service/QuickExportService.java]: Done.

**Review by DmytroBykov1 — APPROVED** [review]

**Pull request merged**